doc/sphinx/addendum/type-classes.rst

```diff
@@ -381,37 +371,65 @@ Summary of the commands
 Besides the :cmd:`Class` and :cmd:`Instance` vernacular commands, there are a
 few other commands related to typeclasses.
 
-.. cmd:: Existing Instance {+ @ident} {? | @natural}
+.. cmd:: Existing Instance @qualid {? @hint_info }
+         Existing Instances {+ @qualid } {? %| @natural }
 
-   This command adds an arbitrary list of constants whose type ends with
+   .. insertprodn hint_info hint_info
+
+   .. prodn::
+      hint_info ::= %| {? @natural } {? @one_term }
+
+   This command adds a constant whose type ends with
    an applied typeclass to the instance database with an optional
    priority :token:`natural`.  It can be used for redeclaring instances at the end of
    sections, or declaring structure projections as instances. This is
    equivalent to ``Hint Resolve ident : typeclass_instances``, except it
    registers instances for :cmd:`Print Instances`.
 
-.. tacn:: typeclasses eauto
-   :name: typeclasses eauto
+.. cmd:: Print Instances @reference
+
+   Shows the list of instances associated with the typeclass :token:`reference`.
+
+
+.. tacn:: typeclasses eauto {? bfs } {? @int_or_var } with {+ @ident }
+          typeclasses eauto {? @int_or_var }
+   :name: typeclasses eauto; _
+
+   .. todo PR: confusing to have both "typeclasses eauto" and "Typeclasses eauto"
```

Comment:
I added wording noting the similar names here and for the command.

As for `bfs`: easily fixed, but not in this PR.  Maybe submit an issue?

I think part of it is that the grammar `TACTIC EXTEND` mechanism isn't as flexible/convenient as productions in `GRAMMAR EXTEND`, so a lot of these enumerate the set of the possible options.  To make the "bfs" optional in a single production, IIUC you have to use an `ARGUMENT EXTEND` to create a new nonterminal, then refer to it as `bfs_opt`.  Same thing for the `"with" ne_preident_list`.

Would be nice if the mechanism could be enhanced so it's more comparable in expressiveness to `GRAMMAR EXTEND`.

```
TACTIC EXTEND typeclasses_eauto
 | [ "typeclasses" "eauto" "bfs" int_or_var_opt(d) "with" ne_preident_list(l) ] ->
    { typeclasses_eauto ~strategy:Bfs ~depth:d l }
 | [ "typeclasses" "eauto" int_or_var_opt(d) "with" ne_preident_list(l) ] ->
    { typeclasses_eauto ~depth:d l }
 | [ "typeclasses" "eauto" int_or_var_opt(d) ] -> {
     typeclasses_eauto ~only_classes:true ~depth:d [Class_tactics.typeclasses_db] }
END
```

Reply:
OK, very interesting explanation! Yes, certainly worth opening an issue (or maybe two issues since one of them could be about the limitations of `TACTIC EXTEND`).

Reply:
While the `TACTIC EXTEND` limitations are a little irritating for developers, they don't block us from doing anything we need to do.  It would be a fair bit of work to improve and wouldn't impact most users, so I don't think I'd choose to work on it unless we find a compelling reason.  In a brief discussion, Hugo was lukewarm.